File: FEBRUARY 24/19 February/Game with String gfg.cpp
class Solution{
public:
    int minValue(string s, int k){
        // code here
        priority_queue<int> pq;
        vector<int> charFreq(26);
        for(char ch:s)
            charFreq[ch-'a']++;
        for(int i=0; i<26; i++)
        {
            if(charFreq[i])
                pq.push(charFreq[i]);
        }
        while(k--)
        {
            int top = pq.top();
            pq.pop();
            if(top-1)
                pq.push(top-1);
        }
        int value=0;
        while(pq.size())
        {
            value+=pow(pq.top(),2);
            pq.pop();
        }
        return value;
    }
};

File: JUNE 2025/26 June/Game with String gfg.java
class Solution {
    public int minValue(String s, int k) {
        // code here
        int[] freq = new int[26];
        for(char ch:s.toCharArray())
        {
            freq[ch-'a']++;
        }
        PriorityQueue<Integer> pq = new PriorityQueue<>(Collections.reverseOrder());
        for(int i=0; i<26; i++)
        {
            if(freq[i]>0)
                pq.offer(freq[i]);
        }
        
        while(k>0)
        {
            int top = pq.poll();
            if(top>1)
                pq.offer(top-1);
            k--;
        }
        int minVal = 0;
        while(!pq.isEmpty())
        {
            int top = pq.poll();
            minVal += top*top;
        }
        return minVal;
    }
}
